### ISSUE TITLE
Duplicate Name in Header

### ISSUE THREAD

**Issue by upsonp** [pyoceans/python-ctd#138]
Was glad to find this project, it should save me a pile of time in reading BTL files.

I ran into an issue with my BTL file header (I don't make them, I just parse them). The Header looks like

```
    Bottle     Bottle        Date Sbeox0ML/L Sbeox1ML/L      Sal00      Sal11 Potemp068C Potemp168C  Sigma-é00  Sigma-é11       Scan      TimeS       PrDM      T068C      C0S/m      T168C      C1S/m       AltM    Par/log    Sbeox0V    Sbeox1V    FlSPuv0       FlSP         Ph TurbWETbb0       Spar   Latitude  Longitude
  Position       S/N         Time                                                                                                                                                                                                                                                                                              
```

When parsed by the _parse_seabird() method this produces names where the first two columns are 'Bottle' and 'Bottle', that causes issues in the df = pd.read_fwf() method causing it to throw a duplicate names error.

Is there a way you'd like to see this resolved that I could put in a PR for?

Maybe append the second row of the header to the first?

**Comment by ocefpaf** [comment]
Do you have a sample file that I can take a look at? I recall seeing something similar but the columns were redundant and we could remove one but it seems that it is not your case here.

**Comment by upsonp** [comment]
I see at the bottom of the ctd.read._parse_seabird() method _remane_duplicate_columns is called to handle this issue (I assume), but it doesn't seem to be functioning correctly.

```
def _remane_duplicate_columns(names):
    items = collections.Counter(names).items()
    dup = []
    for item, count in items:
        if count > 2:
            raise ValueError(
                f"Cannot handle more than two duplicated columns. Found {count} for {item}."
            )
        if count > 1:
            dup.append(item)

    second_occurrences = [names[::-1].index(item) for item in dup]
    for idx in second_occurrences:
        idx += 1
        names[idx] = f"{names[idx]}_"
    return names
```

I would give you a file to test with, but it's Government of Canada data and I don't know what restrictions might be on it. I'll have to ask.

**Comment by upsonp** [comment]
I see at the bottom of the ctd.read._parse_seabird() method _remane_duplicate_columns is called to handle this issue (I assume), but it doesn't seem to be functioning correctly.

```
def _remane_duplicate_columns(names):
    items = collections.Counter(names).items()
    dup = []
    for item, count in items:
        if count > 2:
            raise ValueError(
                f"Cannot handle more than two duplicated columns. Found {count} for {item}."
            )
        if count > 1:
            dup.append(item)

    second_occurrences = [names[::-1].index(item) for item in dup]
    for idx in second_occurrences:
        idx += 1
        names[idx] = f"{names[idx]}_"
    return names
```

I would give you a file to test with, but it's Government of Canada data and I don't know what restrictions might be on it. I'll have to ask.

**Comment by richardsc** [comment]
Hi @upsonp -- if it's a regular AZMP or other monitoring-related file, there shouldn't be an issue with sharing it, as it's all supposed to be public anyway (strictly speaking, all our data is public, just that some PIs will keep it embargoed for a year or two to give time for processing/analysis/publications).

**Comment by upsonp** [comment]
I've been messing around with it and did a bit of research on getting indices in python and I think the issue is with the line

```python
    second_occurrences = [names[::-1].index(item) for item in dup]
```
in the _remane_duplicate_columns() function

if that line is instead

```python
    second_occurrences = [names[1::-1].index(item) for item in dup]
```
then the second occurrence of 'Bottle' gets renamed to 'Bottle_', which I'm pretty sure is what this function is attempting to do.

Now I have a new error that has something to do with date formats...

**Comment by ocefpaf** [comment]
Fixed in #140
